### PR TITLE
Add more testing to more fully show and prove the library's contract

### PR DESCRIPTION
**Pull request by doom-weaver**
### Notes

I've added some test cases to the library to more fully show the contract it meets.

Let me know if you'd like me to add anything.

Thank you for this library.

### Testing

Added the tests you see and executed `yarn test`.

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -51,3 +96,29 @@ test('named export | throw because wrong value type (undefined)', () => {
...
+
+  expect(result).toThrow('value has to be typeof: \'string\' but got typeof: \'object\'');
+});
+
+test('named export | null input throws because wrong value type (object)', () => {
+  const input = null;
+
+  const result = () => convertStringToNumber(input);
+
+  expect(result).toThrow('value has to be typeof: \'string\' but got typeof: \'object\'');
```
Comment (aichbauer) [inline test/index.spec.js]: Just to keep it consistent, store the expected value as a constant.

File: test/index.spec.js [hunk 2]
```diff
@@ -51,3 +96,29 @@ test('named export | throw because wrong value type (undefined)', () => {
...
+
+  expect(result).toThrow('value has to be typeof: \'string\' but got typeof: \'undefined\'');
+});
+
+test('named export | throw because wrong value type (object)', () => {
+  const input = {};
+
+  const result = () => convertStringToNumber(input);
+
+  expect(result).toThrow('value has to be typeof: \'string\' but got typeof: \'object\'');
```
Comment (aichbauer) [inline test/index.spec.js]: Just to keep it consistent, store the expected value as a constant.

File: test/index.spec.js [hunk 3]
```diff
@@ -51,3 +96,29 @@ test('named export | throw because wrong value type (undefined)', () => {
...
   expect(result).toThrow('value has to be typeof: \'string\' but got typeof: \'undefined\'');
 });
+
+
+test('named export | throw because wrong value type (undefined)', () => {
+  const input = undefined;
+
+  const result = () => convertStringToNumber(input);
+
+  expect(result).toThrow('value has to be typeof: \'string\' but got typeof: \'undefined\'');
```
Comment (aichbauer) [inline test/index.spec.js]: Just to keep it consistent, store the expected value as a constant.

File: test/index.spec.js [hunk 4]
```diff
@@ -51,3 +96,29 @@ test('named export | throw because wrong value type (undefined)', () => {
 
   expect(result).toThrow('value has to be typeof: \'string\' but got typeof: \'undefined\'');
```
Comment (aichbauer) [inline test/index.spec.js]: We should also update the value of this test as well.

**Review by aichbauer — CHANGES_REQUESTED** [review]

File: test/index.spec.js (hunk 1 above)
Comment (doom-weaver) [reply test/index.spec.js]: done.

File: test/index.spec.js (hunk 3 above)
Comment (doom-weaver) [reply test/index.spec.js]: done.

File: test/index.spec.js (hunk 4 above)
Comment (doom-weaver) [reply test/index.spec.js]: done.

File: test/index.spec.js (hunk 2 above)
Comment (doom-weaver) [reply test/index.spec.js]: done.

**Pull request closed without merging**